cli/internal/dataplane/write.go

```diff
@@ -251,6 +280,13 @@ func Write(ctx context.Context, uri *url.URL, inputReader io.Reader, options ...
 	return nil
 }
 
+func trackBuffer(window int, buffer *[]byte, bytesRead *int, inputReader io.Reader, err *error) {
+	for {
+		<-time.After(time.Second* time.Duration(window))
```

Comment:
We should be reading as much as possible until `window` elapses, up until we have filled up `blockSize`, not just sleeping for `window`.